src/constants/providers.ts

```diff
@@ -1,14 +1,49 @@
+import { deepCopy } from '@ethersproject/properties'
 // This is the only file which should instantiate new Providers.
 // eslint-disable-next-line @typescript-eslint/no-restricted-imports
 import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { isPlain } from '@reduxjs/toolkit'
 
 import { SupportedChainId } from './chains'
 import { RPC_URLS } from './networks'
 
 class AppJsonRpcProvider extends StaticJsonRpcProvider {
+  private _blockCache = new Map<string, Promise<any>>()
```

Comment:
do you foresee any perf issues here with the size of the payload or anything?

Reply:
I do not, because this is a per-block cache.
If this persisted, I would, but any time it is used a listener is set up to clear the cache at the next block.

Reply:
I hope this is self-documented, by naming it `_blockCache`. There is a comment in the `blockCache` getter, but I don't think add'l doc comments are necessary.